s3repo/controller.py

```diff
@@ -36,6 +38,35 @@ def response_message(message, status):
         response.status_code = status
         return response
 
+    def _generate_repo_annotations(self, path):
+        """Generates a list of repository annotations according to the path."""
+        repo_annotations = []
+        repolist_anchor = 'repolist-'
+
+        # Checking the presence of the "repolist anchor" and that there is only one.
+        if path.find('repolist-') < 0:
+            repo_annotations.append(RepoAnnotation(path, self.model.get_supported_repos()))
+            return repo_annotations
+
+        if len(re.findall(repolist_anchor, path)) != 1:
+            raise RuntimeError('More than one "repolist-" anchor was used.')
+
+        # Extract the name of the repository list and the corresponding list from the config.
+        list_name = re.search(r'{0}(?P<name>[^/]+)'.format(repolist_anchor), path).group('name')
+        repo_list = self.repo_lists.get(list_name, [])
+
+        if not repo_list:
+            raise RuntimeError('List {0} is absent or empty.'.format(list_name))
+
+        # Generates a list of repository annotations.
+        repo_anchor = repolist_anchor + list_name
+        for repo_path in repo_list:
+            generated_path = os.path.normpath(path.replace(repo_anchor, repo_path))
```

Comment:
Here we assume that the replacement ends with an slash, but it is not validated. TBH, I would not require the trailing slash and just do something like `os.path.normpath(path.replace(repo_anchor, repo_path + '/'))`.

Reply:
Hmm, why?
If list name doesn't include the trailing slash:
path=`repolist-modules/el/7`
repo_anchor=`repolist-modules`
list_name=`release/1.10`
generated_path=`release/1.10/el/7`

Am I missing anything?

Reply:
I meant that `repolist-modules-foo/el/7` would be replaced by `release/1.10/-foo/el/7` if I undestood the code right.

Reply:
Updated.